### ISSUE TITLE
After moving from sqlite-net-pcl to sqlite-net-sqlcipher, iOS wont build

### ISSUE THREAD

**Issue by Sebastian1989101** [praeclarum/sqlite-net#836]
After removing sqlite-net-pcl (1.5.231) and adding sqlite-net-sqlcipher (1.5.231) I'm no longer able to build my project. I always get this message:

`error MT2101: Can't resolve the reference 'System.Void SQLite.SQLiteConnection::.ctor(System.String,System.Boolean,System.Object)', referenced from the method 'SQLite.SQLiteConnection MyApp.iOS.DependencyServices.SQLiteService::GetConnection(MyApp.Core.DependencyServices.Database,System.String)' in 'SQLite-net, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null'.`

If I add sqlite-net-pcl side-by-side with the sqlcipher version, the app is using the wrong method and gets a "file is not a database" error. The very same projects runs fine on Android (identical code). So what am I missing here? My iOS project currently references `sqlite-net-sqlcipher (1.5.231)`, `SQLitePCLRaw.bundle_sqlcipher (1.1.13)` and `SQLitePCLRaw.bundle_green (1.1.13)`.

**Comment by ericsink** [comment]
I was going to suggest removing bundle_green, but you did that already.

The remaining error is weird.  Have you by chance done a full clean/rebuild of your project?

**Comment by Sebastian1989101** [comment]
@ericsink Yes. I also deleted every obj and bin folder manual while the VS for Mac was closed. The error is thrown in the .NET Standard 2.0 core project which only has a reference to `sqlite-net-sqlcipher (1.5.231)` this one has a reference to `SQLitePCLRaw.bundle_sqlcipher (1.1.11)` internally but even if I install 1.1.13 manual of this to the core project, the error remains. 

My project also uses Microsoft.AppCenter and Mapsui which both using SQLite internally - is this maybe the problem? Because I don't know the versions used inside of them but I know both using SQLite. 

I have no idea anymore what could cause the issue because the Android project runs fine and this one has references to `sqlite-net-sqlcipher (1.5.231)`, `SQLitePCLRaw.bundle_sqlcipher (1.1.13)` and `SQLitePCLRaw.bundle_green (1.1.13)` just like the iOS project had in the beginning...

**Comment by ericsink** [comment]
Ah yes, AppCenter.

You might want to look through this issue:

https://github.com/ericsink/SQLitePCL.raw/issues/169

The excludeAssets trick should help.

**Comment by Sebastian1989101** [comment]
@ericsink Do I have to add the excludeAssets to AppCenter or to the SQLite libraries? 

Because as soon as I change this: 

```
    <PackageReference Include="sqlite-net-sqlcipher">
      <Version>1.5.231</Version>
    </PackageReference>
    <PackageReference Include="SQLitePCLRaw.bundle_green">
      <Version>1.1.13</Version>
    </PackageReference>
    <PackageReference Include="SQLitePCLRaw.bundle_sqlcipher">
      <Version>1.1.13</Version>
    </PackageReference>
```

to this: 

```
    <PackageReference Include="sqlite-net-sqlcipher">
      <Version>1.5.231</Version>
      <ExcludeAssets>All</ExcludeAssets>
    </PackageReference>
    <PackageReference Include="SQLitePCLRaw.bundle_sqlcipher">
      <Version>1.1.13</Version>
      <ExcludeAssets>All</ExcludeAssets>
    </PackageReference>
```

in my iOS project only, I can't compile anymore because the `new SQLiteAsyncConnection(string, string)` throws a build error: `The best overload for 'SQLiteAsyncConnection' does not have a parameter named 'key'`

**Comment by ericsink** [comment]
AppCenter is bringing in conflicting stuff you don't want.  The excludeAssets trick is just a way to get that stuff ignored.

You definitely don't want to excludeAssets for bundle_sqlcipher, since that's the one you want.

You definitely DO want to excludeAssets for bundle_green, since IIRC that's the one AppCenter is bringing in, and you can't have two bundles.

These excludeAssets thing should go in the csproj for your top-level app project (iOS).

**Comment by Sebastian1989101** [comment]
@ericsink So in this case if I understand it correctly, I need to add `<ExcludeAssets>All</ExcludeAssets>` to the bundle_green which I completly removed anyway?

**Comment by ericsink** [comment]
What I'm suggesting you try is add the following to your csproj file:

    <PackageReference Include="SQLitePCLRaw.bundle_green">
      <Version>1.*</Version>
      <ExcludeAssets>All</ExcludeAssets>
    </PackageReference>

Which means:

"If any of my dependencies tries to bring in bundle_green, prevent that from happening."

**Comment by Sebastian1989101** [comment]
@ericsink See edit above. Same error when I add this.

**Comment by ericsink** [comment]
Hmmm.  Then for the moment I am stumped.  Situations involving AppCenter can be rather problematic.

If you want to try something else:

In your portable library, instead of using sqlite-net-sqlcipher, use sqlite-net-base, which is the same code except without a dependency on a bundle.

Then, in your iOS app project, reference bundle_sqlcipher.

And do keep the excludeAsset stuff you've got now, to prevent AppCenter from bringing in sqlite-net or bundle_green.

If you follow these steps, you'll probably then need to manually initial the bundle by adding a call to `SQLitePCL.Batteries_V2.Init()` in your app startup code.

**Comment by Sebastian1989101** [comment]
@ericsink This seems to work. Now I only get a "database is locked" error when I start the app a 2nd time... so I guess I need to find the part now where it does not close properly... Is there some kind of different behavior between the unencrypted and the cipher version?

**Comment by ericsink** [comment]
With regard to locking, no, nothing I can think of.
